segmentation/utils/segmentation_utils.py

```diff
@@ -16,7 +16,110 @@
+        tiff_dir (str): the name of the directory which contains the single_channel_inputs,
+            technically can be accessed from base_dir but easier to make its own separate arg
+        img_sub_folder (str): the name of the folder where the TIF images are located
+        points (list): a list of points we wish to analyze, if None will default to all points
+        is_mibitiff (bool): a flag to indicate whether or not the base images are MIBItiffs
+        mibitiff_suffix (str): if is_mibitiff is true, then needs to be specified to select
+            which points to load from mibitiff
+        segmentation_labels (xarray): an xarray with the segmented data
+        batch_size (int): how large we want each of the batches of points to be when computing,
+            adjust as necessary for speed and memory considerations
+
+    Returns:
+        combined_normalized_data (pandas): a DataFrame containing the data with the size_norm transformation
+        combined_transformed_data (pandas): a DataFrame containing the data with the arcsinh transformation
+    """
+
+    if not base_dir:
+        raise ValueError("base_dir not specified")
+
+    if not tiff_dir:
+        raise ValueError("tiff_dir not specified")
+
+    if not img_sub_folder:
+        raise ValueError("img_sub_folder not specified")
+
+    # if no points are specified, then load all the points
+    if not points:
+        # if MIBItiff, only look at the points with the desired suffix...
+        if is_mibitiff:
+            # ...but of course, check if that suffix has been defined first
+            if not mibitiff_suffix:
+                raise ValueError("No mibitiff_suffix specified: please do so if loading from MIBItiff")
+
+            points = [point for point in points if point.endswith(MIBItiff_suffix)]
+        else:
+            # load channel data
+            all_points = os.listdir(tiff_dir)
+            all_points = [point for point in all_points if os.path.isdir(os.path.join(tiff_dir, point))]
+        points = all_points
+
+    # sort the points
+    points.sort()
+
+    # defined some vars for batch processing
+    cohort_len = len(points)
+    num_batch = int(np.floor(cohort_len / batch_size))
+
+    # create the final dfs to store the processed data
+    combined_normalized_data = pd.DataFrame()
+    combined_transformed_data = pd.DataFrame()
+
+    # iterate over all the batches
+    for i in range(num_batch):
+        # extract only the points we need for the current batch
+        current_points = points[i * batch_size:(i + 1) * batch_size]
+
+        # assert that all the current_points specified appear as fovs in segmentation_labels
+        # we need this check because otherwise load_imgs_from_tree will fail when we try to
+        # validate the paths because they would not exist to start off with
+        if not np.all(np.in1d(np.array(current_points), segmentation_labels['fovs'].values)):
```

Comment:
Rather than doing this inside the for loop, we should do this at the beginning. 

We should also print out the specific values that are missing

Reply:
Ah yes, thanks for pointing that out.